featurebyte/service/feature_table_cache.py

```diff
@@ -1,111 +1,448 @@
-):
-    """
-    Feature Table Cache metadata service
+def alter_table_add_columns_sql(
+    table: expressions.Table,
+    columns: List[expressions.ColumnDef],
+    source_type: SourceType,
+) -> str:
+    """Generate ALTER TABLE ADD COLUMN sql.
+    sqlglot doesn't support it well at the moment of writing this code
+    - for snowflake only 1 column alter commands work, multiple columns result into invalid sql
+    - for spark any sql generated by sqlglot is invalid
+
+    Parameters
+    ----------
+    table: expressions.Table
+        Table to alter
+    columns: List[expressions.ColumnDef]
+        List of columns to add
+    source_type: SourceType
+        DWH Source type
+
+    Returns
+    -------
+    str
+        ALTER sql statement
     """
+    alter_table_sql = f"ALTER TABLE {sql_to_string(table, source_type=source_type)}"
+
+    if get_dialect_from_source_type(source_type) == "snowflake":
+        first = sql_to_string(columns[0], source_type=source_type)
+        rest = ",\n".join(
+            [sql_to_string(col, source_type=source_type) for col in columns[1:]]
+        ).strip()
+        alter_table_sql += f" ADD COLUMN {first}"
+        if rest:
+            alter_table_sql += f",\n{rest}"
+    else:
+        tuple_expr = expressions.Tuple(expressions=columns)
+        alter_table_sql += " ADD COLUMNS " + sql_to_string(tuple_expr, source_type=source_type)
 
-    document_class = FeatureTableCacheModel
+    return alter_table_sql
+
+
+class FeatureTableCacheService:
+    """
+    Service for managing physical feature table cache as well as metadata storage.
+    """
 
     def __init__(
         self,
-        user: Any,
-        persistent: Persistent,
-        catalog_id: Optional[ObjectId],
-        block_modification_handler: BlockModificationHandler,
-        observation_table_service: ObservationTableService,
+        feature_table_cache_metadata_service: FeatureTableCacheMetadataService,
+        namespace_handler: NamespaceHandler,
+        session_manager_service: SessionManagerService,
+        entity_validation_service: EntityValidationService,
+        tile_cache_service: TileCacheService,
+        feature_list_service: FeatureListService,
     ):
-        super().__init__(
-            user=user,
-            persistent=persistent,
-            catalog_id=catalog_id,
-            block_modification_handler=block_modification_handler,
-        )
-        self.observation_table_service = observation_table_service
+        self.feature_table_cache_metadata_service = feature_table_cache_metadata_service
+        self.namespace_handler = namespace_handler
+        self.session_manager_service = session_manager_service
+        self.entity_validation_service = entity_validation_service
+        self.tile_cache_service = tile_cache_service
+        self.feature_list_service = feature_list_service
+
+    def create_feature_name(self, definition_hash: str) -> str:
```

Comment:
Is this referring to the column name in the cache table, not really the name of any features? I feel it is less confusing if we call this `column_name` instead of `feature_name`.

Reply:
Besides, would prefer to move this logic to the model level rather than service level as:
* it is easier to test (without the need to construct the service)
* personally, i would search the models first to check where the name of the object is constructed rather than in service level modules.

Reply:
moving to model is a good idea 👍